contracts/SL2RD.sol

```diff
@@ -152,6 +158,108 @@ contract SL2RD is
         setInitialized();
     }
 
+    /// @notice Prepares the contract for multipart initialization.
+    /// @dev Recipient tokenId table is constructed off-chain and is
+    /// constructed as follows:
+    /// 1. A number of slots are allocated in an array corresponding
+    /// to 1 / minimum percentage of any stakeholder in this asset.
+    /// 2. IDs are assigned to slots redundantly such that the probability
+    /// of an iterator pointing to a given ID is equal to that
+    /// ID's ownership stake in the asset.
+    /// 3. Additionally, the layout specified in (2) is randomized
+    /// such that primary stakeholders have no advantage in payment
+    /// time given that the iterator increments linearly through the
+    /// ID table.
+    /// 4. A parallel addresses array is filled with the corresponding
+    /// owner for each tokenId. This is prefilled with the owner's
+    /// address for each token, but designed with flexibilty to include
+    /// others during initialization if they are known.
```

Comment:
@un7c0rn didn't we find an issue during one of our deployments that required all splits, to be transferrable, to be initialized to the contract owner's address? Or is that logic within the transfer RPC and not the smart contract?

Reply:
@kim-formless Yes I think you're correct and in looking at the code I believe the issue is that _nextAvailableCommunitySlot is unconditionally initialized to 0 rather than to a value which reserves some pre-allocated memory. In theory the purpose of line 4 here is in the case where some portion of the distribution table has been reserved for _non_ community splits, e.g. [ 60% CREATORS ] [ 40% COMMUNITY ] in which case the initial value of "next available community slot" would be slot 60 and not slot 0.

Reply:
This is what I was referring to here, https://github.com/formless-eng/share-v1-core/pull/42#issuecomment-1863780391 :)

Reply:
@Jpsassine Nice! Just left a few notes and questions on that PR.

Reply:
@kim-formless In the interim I've tracked here to decouple these two issues: https://github.com/formless-eng/share-v1-core/issues/50

Reply:
> @kim-formless @un7c0rn I believe this is what I was referring to here, [#42 (comment)](https://github.com/formless-eng/share-v1-core/pull/42#issuecomment-1863780391)

@Jpsassine nice, and hi John Paul!!

Reply:
> @kim-formless In the interim I've tracked here to decouple these two issues: #50

@un7c0rn thanks, and SGTM!